services/web/server/src/simcore_service_webserver/projects/projects_handlers.py

```diff
@@ -125,12 +128,30 @@ async def list_projects(request: web.Request):
     db = request.config_dict[APP_PROJECT_DBAPI]
 
     # TODO: improve dbapi to list project
+    async def set_all_project_states(projects: List[Dict[str, Any]], is_template: bool):
+        await logged_gather(
+            *[
+                projects_api.add_project_states_for_user(
+                    user_id=user_id,
+                    project=prj,
+                    is_template=is_template,
+                    app=request.app,
+                )
+                for prj in projects
+            ],
+            reraise=True,
```

Comment:
so an error to add states in project ends in a 505 for the front-end?
Can you guarantee atomicity in this change or you can have some projects updated and others not? is there any side effect or next request will fix it?

Reply:
ok, you have a point here. I did not change the initial behavior. the get_computation_task handles partly these things. but it is true that if there is a network glitch this would end up in a 500 which is bad.

getting the states means:
1. accessing redis to get the ProjectLockState -> this could faill and is currently raising some redis exception which would end in a 500. But this is probably not new. -> I'll create a new issue for it if that is ok with you.
2. accessing director-v2 to get the computation pipeline status -> anything there is handled and sets the state to UNKNOWN but if there is a timeout issue. then a 503 comes back.

I need to think about it